packages/graphback-datasync/src/deltaHelper.ts

```diff
@@ -0,0 +1,77 @@
+import { ModelDefinition, getTableName } from '@graphback/core';
+import { Db, Collection } from "mongodb";
+import { DataSyncFieldNames } from './util';
+
+export function getDeltaTableName(tableName: string) {
+  return `${tableName}Delta`
+}
+
+/**
+ * M
+ */
+export class MongoDeltaHelper {
+  protected collection: Collection;
+  public constructor(model: ModelDefinition, db: Db) {
+    this.collection = db.collection(getDeltaTableName(getTableName(model.graphqlType)));
+  }
+
+  public async insertDiff(updateDocument: any) {
+    const { _id,  ...delta } = updateDocument;
+    const version = delta[DataSyncFieldNames.version];
+    // eslint-disable-next-line @typescript-eslint/tslint/config
+    delete delta[DataSyncFieldNames.version];delete delta[DataSyncFieldNames.lastUpdatedAt];delete delta[DataSyncFieldNames.deleted]; 
+
+    const diff: any = {
+      docId: _id,
+      [DataSyncFieldNames.version]: version,
+      delta
+    }
+    console.log(diff);
+    await this.collection.insertOne(diff);
+  }
+
+  // returns final resolved update document
+  public async resolveConflictFromDelta(updateDocument: any): Promise<any> {
```

Comment:
Resolving conflict algorithm for dummies.
```
1. Accept client side version form client and validate it.
2. Look for the delta table entry with id and version.
3. This entry will be our `base` for conflict resolution.
4. Fetch server side entry 
5. Calculate diff for server side entry
5. Perform conflict check (check if fields that were changed are mergeable based on the diffs)
6. if mergable perform merge
7. If not mergable (the same fields changed - perform predefined conflict resolution.
8. Call out resolution algorithm (user provided function or default ones we provide)
We can use `MostRecentWins strategy` - that is why do not remove last updated :)
9.  try to save result to database - conditial check if version did not changed since then
7. Return data to user.

Another condition - if items reached their time (TTL ;) we have no way to kick offix compatible error to user.

```

All of this logic is already implemented in:
https://github.com/aerogear/offix/tree/master/packages/offix-conflicts-client

See also offix conflict server. Obviously you are free to invent your own algorithm but based on what I see I will probably prefer us to copy paste the code that was written and used in production rather than try to invent new way :)